step-templates/azure-check-vmss-provision-status.json

```
@@ -0,0 +1,138 @@
+        "HelpText": "**Required**\n\nThe Azure Account to use when querying the VMSS.",
+        "DefaultValue": "",
+        "DisplaySettings": {
+          "Octopus.ControlType": "AzureAccount"
+        }
+      },
+      {
+        "Id": "b03d1bf3-76f8-4a3c-917a-60d5e19a15f4",
+        "Name": "VMSS.DeploymentTarget.Roles",
+        "Label": "Deployment Target Roles",
+        "HelpText": "**Required**\n\nA comma-separated list of deployment target roles to filter your deployment targets by.  These roles are how this step will determine which machines to reconcile.\n\nIf you supply multiple roles, for example `todo-web-server,todo-virtual-machine-scale-set` the deployment targets have to be assigned `todo-web-server` AND `todo-virtual-machine-scaleset` roles for the target to be considered.",
+        "DefaultValue": "",
+        "DisplaySettings": {
+          "Octopus.ControlType": "SingleLineText"
+        }
+      },
+      {
+        "Id": "757ca61b-d15b-4dbb-911a-b3f4dc0258b5",
+        "Name": "VMSS.Octopus.Url",
+        "Label": "Octopus URL",
+        "HelpText": "**Required**\n\nThe URL of the Octopus Server to query against.  Example: `https://samples.octopus.app`.",
+        "DefaultValue": "#{Octopus.Web.ServerUri}",
+        "DisplaySettings": {
+          "Octopus.ControlType": "SingleLineText"
+        }
+      },
+      {
+        "Id": "7b070182-e79b-48a0-966a-39d332802b60",
+        "Name": "VMSS.Octopus.ApiKey",
+        "Label": "Octopus API Key",
+        "HelpText": "**Required**\n\nAPI Key of a service account that has permissions to:\n- Query deployment targets, environments, and tenants. \n- Query events (audit history) \n- Delete deployment targets\n\nAssigning the service account to a team with `Project Viewer` and `Environment Manager` roles will work.",
+        "DefaultValue": "",
+        "DisplaySettings": {
+          "Octopus.ControlType": "Sensitive"
+        }
+      },
+      {
+        "Id": "6e0f8635-70ba-4b8a-b5b4-909dba3f9fd8",
+        "Name": "VMSS.Timeout.Value",
+        "Label": "Timeout (In Minutes)",
+        "HelpText": "**Required**\n\nHow long this step will wait (in minutes) for the VMSS and VMs in the VMSS to finish being created.",
+        "DefaultValue": "30",
+        "DisplaySettings": {
+          "Octopus.ControlType": "SingleLineText"
+        }
+      },
+      {
+        "Id": "0392f153-73b0-44d4-9181-4f366c9012f7",
+        "Name": "VMSS.Timeout.ErrorHandle",
+        "Label": "Timeout Error",
+        "HelpText": "**Required**\n\nWhat will happen when a timeout occurs.\n\n- `Proceed`: The script will reconcile the VMs it can and then finish.  No error thrown.  \n- `Error`: Will throw an error and it will stop the deployment from proceeding.\n\nThe default is `Proceed`",
+        "DefaultValue": "Proceed",
+        "DisplaySettings": {
+          "Octopus.ControlType": "Select",
+          "Octopus.SelectOptions": "Proceed|Proceed\nCancel|Cancel"
```

The script depends on this parameter having a value of `Error`. The Option in the dropdown here is `Cancel`. Looking at the script, I think this line needs to change to:

```suggestion
          "Octopus.SelectOptions": "Proceed|Proceed\nError|Error"
```

In addition, the script has this line in it (L258):

`if ($timeoutErrorHandle -eq "Error")`

I think it should be changed to:

`if ($timeoutErrorHandle.ToLower().Trim() -eq "error")`

This will make it consistent with the other checks

---

There are two drop-down lists at play here, and I think there is some confusion.


Cancel is for a duplicate run (VMSS.Duplicate.Handle)

```
f ($duplicateRunHandle.ToLower().Trim() -eq "cancel")
            {
            	Write-Highlight "The duplicate run handle is set to cancel, cancelling current deployment."
                Invoke-OctopusApi -octopusUrl $octopusUrl -apiKey $apiKey -spaceId $OctopusSpaceId -method "POST" -endPoint "tasks/$($octopusTaskId)/cancel"    
            }
```

The timeout is the one with the error (VMSS.Timeout.ErrorHandle)

```
if ($timeoutErrorHandle.ToLower().Trim() -eq "error")
            {
            	Write-Highlight "The VMs in the scale have been provisioning for over $timeoutInMinutes.  Error handle is set to error out, exiting with an exit code of 1."
                exit 1
            }
```

---

The Timeout Error handler parameter in the JSON is the one that currently has a dropdown option of Cancel, and no `Error` option is provided. So I don't think this would work as intended.

---

Ah, good catch, updating